ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java

```diff
@@ -25,6 +25,9 @@
 
 /** An asynchronous output stream supporting zero buffer copying. */
 public interface DataStreamOutput extends CloseAsync<DataStreamReply> {
+  /** Create a RaftClientRequest and send out it asynchronously */
+  CompletableFuture<DataStreamReply> streamRequestAsync();
+
   /** Send out the data in the buffer asynchronously */
-  CompletableFuture<DataStreamReply> streamAsync(ByteBuffer buf);
+  CompletableFuture<DataStreamReply> streamDataAsync(ByteBuffer buf);
```

Comment:
Let's use the name "writeAsync".  It is consistent with the WritableByteChannel.write(ByteBuffer) method.